src/devices/fakebike/fakebike.cpp

```diff
@@ -163,4 +165,35 @@ void fakebike::changeInclinationRequested(double grade, double percentage) {
     changeInclination(grade, percentage);
 }
 
+uint16_t fakebike::wattsFromResistance(double resistance) {
+    return _ergTable.estimateWattage(Cadence.value(), resistance);
+}
+
+resistance_t fakebike::resistanceFromPowerRequest(uint16_t power) {
+    //QSettings settings;
+    //bool toorx_srx_3500 = settings.value(QZSettings::toorx_srx_3500, QZSettings::default_toorx_srx_3500).toBool();
+    /*if(toorx_srx_3500)*/ {
+        qDebug() << QStringLiteral("resistanceFromPowerRequest") << Cadence.value();
+
+        if (Cadence.value() == 0)
+            return 1;
+
+        for (resistance_t i = 1; i < maxResistance(); i++) {
+            if (wattsFromResistance(i) <= power && wattsFromResistance(i + 1) >= power) {
```

Comment:
This function could be offered by the ergTable class. 

Also, this is doing the ergTable search twice for almost every resistance level from 1 to the result. You could save the computation by caching the values.
e.g.:
```
uint16_t power1 = wattsFromResistance(1), lastPower = power1;
for(resistance_t i = 1; i<maxResistance(); i++) {
   uint16_t thisPower = wattsFromResistance(i+1);
   if(lastPower <= power && thisPower>=power) {
        qDebug() << QStringLiteral("resistanceFromPowerRequest") << lastPower;
        return i;
   }
   lastPower = thisPower;  
}
return power < power1 ? 1:maxResistance();
```